src/main/java/com/simibubi/create/compat/jei/category/CreateRecipeCategory.java

```diff
@@ -124,7 +124,7 @@ public static IRecipeSlotRichTooltipCallback addStochasticTooltip(ProcessingOutp
 	@SuppressWarnings("removal") // see below
 	public static IRecipeSlotBuilder addFluidSlot(IRecipeLayoutBuilder builder, int x, int y, FluidIngredient ingredient) {
 		int amount = ingredient.getRequiredAmount();
-		return builder.addSlot(RecipeIngredientRole.OUTPUT, x, y)
+		return builder.addSlot(RecipeIngredientRole.INPUT, x, y)
 			.setBackground(getRenderedSlot(), -1, -1)
 			.addIngredients(ForgeTypes.FLUID_STACK, ingredient.getMatchingFluidStacks())
 			.setFluidRenderer(amount, false, 16, 16) // make fluid take up the full slot
```

Comment:
Using 1 to replace the `amount` is sufficient enough to make the fluid take up the full slot (JEI will limit the rendered fluid not exceeding the height limit), which should eliminate the need to pass in the fluid ingredient or fluid stack to get the amount, and allows rendering multiple fluid stacks that might not be of the same size in the slot, the current methods can still be left as short cuts.